content/sips/sip-230.md

```diff
@@ -12,41 +12,33 @@ created: 2022/04/07
 
-Implementation of SIP-165, and to some extent, futures, have revealed that we need a more self-contained, generic system for chainlink oracle validation is required. To solve this problem,
-a new circuit breaker contract which all queries to chainlink aggregators must run through is created.
-
-For testing purposes, this class can be used to reduce the provisioning complexity of the system in general by being able to set override prices from the
-circuit breaker system itself.
+Implementation of SIP-165, and to some extent, futures, have revealed that we need a more generic system for chainlink oracle validation is required. To solve this problem, an update to the existing circuit breaker contract will allow tracking for any oracle. In addition, the circuit breaker system
+is designed to be more self contained to allow for arbitrary oracles to be supported without needing to integrate with `SystemStatus` directly.
 
 ## Motivation
 
 <!--This is where you explain the reasoning behind how you propose to solve the problem. Why did you propose to implement the change in this way, what were the considerations and trade-offs? The rationale fleshes out what motivated the design and why particular design decisions were made. It should describe alternate designs that were considered and related work. The rationale may also provide evidence of consensus within the community, and should discuss important objections or concerns raised during discussion.-->
 
-This is a technical debt item which will improve the maintainability and security of the protocol
+This is a technical debt item which will improve the maintainability and security of the protocol.
+
+Additionally, this will make it significantly easier to have full coverage of all oracles in use in our system. Currently, the SNX price oracle and
+the `IssuedSynths` oracle are not covered by any validation. We can now close the holes for those oracles with this.
 
 ### Technical Specification
 
-#### `OracleManager`
+#### `CircuitBreaker`
 
-This is a new contract added to the system which will replace `ExchangeCircuitBreaker`, portions of `ExchangeRates` where aggregators are stored,
-and any other circuit breaker functionality which previously existed will be removed and replaced with a call to this contract.
+A new copy of `ExchangeCircuitBreaker` which is specifically designed to handle any type of oracle value--not just the ones used by exchange rate. Broadly speaking, the contrat will be changed to 1) store internal values for broken circuit status rather than updating `SystemStatus` and, 2) accept `address oracleAddress` rather than `bytes32 currencyKey` in order to allow for any oracle to be submitted, and 3) integrate these checks into `ExchangeRates` rather than the othre way around (`ExchangeCircuitBreaker` calls `ExchangeRates`).
```

Comment:
> store internal values for broken circuit status rather than updating `SystemStatus`

maybe use flexibleStorage in order to prevent the need for complex migration flows if something is paused?

Reply:
`FlexibleStorage` is not worth it in this case. If we do a system upgrade of `CircuitBreaker` and a circuit is broken while that happens (a case that should be infrequent and quickly resolved after the pdao has done its investigation of the oracle), the pdao can choose to suspend the synth if there is a problem, or force a trigger of the breaker by setting the last oracle value to a bogus number.